src/graphql/directives/scheduled-sections.js

```diff
@@ -0,0 +1,23 @@
+const { SchemaDirectiveVisitor } = require('graphql-tools');
+
+class ScheduledSectionsDirective extends SchemaDirectiveVisitor {
+  /**
+   *
+   * @param {*} field
+   */
+  visitFieldDefinition(field) { // eslint-disable-line class-methods-use-this
+    // eslint-disable-next-line no-param-reassign
+    field.resolve = async ({ _id }, { input: { pagination, sort } }, { base4 }) => {
+      const coll = await base4.collection('website', 'Schedule');
+      const ids = await coll.distinct('section', { 'content.$id': _id });
```

Comment:
I would create a `base4.distinct` method that retrieves the collection and passes the criteria under the hood...
```js
async distinct(modelName, field, query) {
  const { namespace, resource } = Base4.parseModelName(modelName);
  const collection = await this.collection(namespace, resource);
  return collection.distinct(field, query);
}